instruqt-tracks/nomad-integration-with-vault/track.yml

```diff
@@ -0,0 +1,490 @@
+      CREATE USER "{{name}}" WITH ENCRYPTED PASSWORD '{{password}}' VALID UNTIL '{{expiration}}';
+      ALTER USER "{{name}}" WITH SUPERUSER;
+      </pre>
+      <p>The SQL above will be used in the creation_statements parameter of our next command to specify the privileges that the dynamic credentials being generated will possess. In our case, the dynamic database user will have broad privileges that include the ability to read from the tables that our application will need to access.</p>
+  - type: text
+    contents: |-
+      <p>Finally, we need to specify a policy named access-tables in our allowed_policies section of the token role. We will create this policy now and give it the capability to read from the database/creds/accessdb endpoint. We will then specify this policy in our Nomad job which will allow it to retrieve credentials for itself to access the database.</p>
+      <pre style="overflow-x: auto; white-space: pre-wrap; white-space: -moz-pre-wrap; white-space: -pre-wrap; white-space: -o-pre-wrap; word-wrap: break-word; padding: 15px;">
+      path "database/creds/accessdb" {
+        capabilities = ["read"]
+      }
+      </pre>
+  tabs:
+  - title: Server
+    type: terminal
+    hostname: hashistack-server
+  - title: Files
+    type: code
+    hostname: hashistack-server
+    path: /root/hashistack/
+  - title: Nomad
+    type: service
+    hostname: hashistack-server
+    port: 4646
+  - title: Consul
+    type: service
+    hostname: hashistack-server
+    port: 8500
+  - title: Vault
+    type: service
+    hostname: hashistack-server
+    port: 8200
+  difficulty: basic
+  timelimit: 300
+- slug: deploy-an-application
+  id: 4ddvvt3xlnzr
+  type: challenge
+  title: Deploy an Application
+  teaser: Deploy an Application with the Appropriate Policy and Templating
+  assignment: |-
+    The Nomad job file has been also been created for you and you can take a look at it by clicking on the <b>"Files"</b> tab and navigating to <b>hashistack > vault > web-app.nomad</b>.</ br></ br>
+    Run the Nomad job by running the command;
+    ```
+    nomad run /root/hashistack/nomad/web.nomad
+    ```
+    Click on the <b>"Nomad"</b> tab and wait until the job has been deployed and healthy.</ br></ br>
+    Confirm that the application is accessing the database by running two commands. Use the dig command to query the SRV record of your service and obtain the port it is using;
+    ```
+    dig +short SRV web.service.consul.
+    ```
+    The response should look similar to <b>1 1 30478 ip-172-31-58-230.node.instruct.consul.</b>. Next curl your service at the appropriate port and names path;
+    ```
+    curl http://web.service.consul:3000/api
+    ```
+    The response should return json from the api.<br /><br />
+    Finally, click of the <b>"Application"</b> tab to view the web application.<br /><br />
```

Comment:
When looking at the app, I was left wondering how it was using the database.  I imagine all the content in the screen came from the database.  But it might be worth pointing that out.  It would also be good to show what the contents of the secrets/file.env that is provided to the application looks like and show a snippet of the app's code that reads that file.

Reply:
I tried viewing the file in the Nomad UI, but it is an unsupported file type for the UI.  When I clicked on the "View Raw File" button, Instruqt tried to launch a new tab outside the tab that had the track, but I saw a "Connecting to service ..." message.  perhaps it could be shown via Nomad CLI commands?

Reply:
Unfortunately, running `nomad alloc fs 966c2e3f /server/secrets/file.env` gives me "Error reading file: Unexpected response code: 500 (Reading secret file prohibited: /server/secrets/file.env)"

Reply:
And I don't think that `nomad alloc exec 966c2e3f ls` will give access to that directory with the secrets since it lists:
Dockerfile         node_modules       public
app.js             package-lock.json  routes
bin                package.json       views